### PR TITLE
[MapView][Part 4] Make BaseMapView own rendering-related configuration options.

### PR DESCRIPTION
**Pull request by nishant-karajgikar**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [x] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: `<changelog></changelog>`.
 - [ ] Update the migration guide, API Docs, Markdown files - Readme, Developing, etc

### Summary of changes

This PR makes the `BaseMapView` own the rendering-related options that used to be present in `MapConfig`, i.e., the `RenderOptions`.

### CODE REVIEW

File: Apps/Examples/Examples/All Examples/RestrictCoordinateBoundsExample.swift [hunk 1]
```diff
@@ -22,7 +21,7 @@ public class RestrictCoordinateBoundsExample: UIViewController, ExampleProtocol
 
         // Center the camera on the bounds
         let camera = mapView.mapboxMap.camera(for: bounds, padding: .zero, bearing: 0, pitch: 0)
-        
+
         // Set the camera's center coordinate.
```
Comment (neelmistry94) [inline Apps/Examples/Examples/All Examples/RestrictCoordinateBoundsExample.swift:25]: Nits, looks like there were no real changes here so can they be removed so there is not change to this file

File: Sources/MapboxMaps/Foundation/BaseMapView.swift [hunk 2]
```diff
@@ -32,6 +32,15 @@ open class BaseMapView: UIView {
 
     private let mapClient = DelegatingMapClient()
 
+    public var options = RenderOptions() {
+        didSet {
+            let defaultPrefetchZoomDelta: UInt8 = 4
```
Comment (neelmistry94) [inline Sources/MapboxMaps/Foundation/BaseMapView.swift:37]: Is this a known constant that needs to be used outside of this scope? If so, we shouldn't make the variable available, only in the didSet local scope. 

Also, what is the context around the value of 4? Can we add some documentation around this please?

File: Sources/MapboxMaps/Foundation/BaseMapView.swift [hunk 3]
```diff
@@ -127,6 +136,14 @@ open class BaseMapView: UIView {
         if let cameraOptions = resolvedMapInitOptions.cameraOptions {
             mapboxMap.__map.setCameraFor(MapboxCoreMaps.CameraOptions(cameraOptions))
         }
+
+        // Set prefetchZoomDelta
+        let defaultPrefetchZoomDelta: UInt8 = 4
```
Comment (neelmistry94) [inline Sources/MapboxMaps/Foundation/BaseMapView.swift:141]: Similar comment to above, Looks like this same value is being reused so we should make this a private constant in the class

File: Sources/MapboxMaps/Foundation/BaseMapView.swift [hunk 4]
```diff
@@ -127,6 +136,14 @@ open class BaseMapView: UIView {
...
+        mapboxMap.__map.setPrefetchZoomDeltaForDelta(
+            options.prefetchesTiles ? defaultPrefetchZoomDelta : 0)
```
Comment (neelmistry94) [inline Sources/MapboxMaps/Foundation/BaseMapView.swift:143]: Weird spacing, put it all on line 142

File: Sources/MapboxMaps/Foundation/BaseMapView.swift (hunk 3 above)
Comment (nishant-karajgikar) [reply Sources/MapboxMaps/Foundation/BaseMapView.swift:143]: There's a ticket open for this. We should address it as part of this.

File: Sources/MapboxMaps/Foundation/BaseMapView.swift (hunk 2 above)
Comment (nishant-karajgikar) [reply Sources/MapboxMaps/Foundation/BaseMapView.swift:37]: There's a ticket open for this.

File: Sources/MapboxMaps/Foundation/BaseMapView.swift (hunk 4 above)
Comment (neelmistry94) [reply Sources/MapboxMaps/Foundation/BaseMapView.swift:145]: hmmm interesting. But that's fine, its a nit

**Review by neelmistry94 — APPROVED** [review]

**Pull request merged**